crates/ruff_linter/src/linter.rs

```diff
@@ -416,6 +424,12 @@ pub fn add_noqa_to_path(
         target_version,
     );
 
+    let suppressions = if is_range_suppressions_enabled(settings) {
+        Suppressions::from_tokens(locator.contents(), parsed.tokens())
+    } else {
+        Suppressions::default()
+    };
+
```

Comment:
We now end up extracting the suppressions twice: Once in `check_path` and once here. We might need to pass the suppressions as part of `check_path`